src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from "react";
+import { MovieCard } from "../movie-card/movie-card";
+import { Form, Button, Row, Col, Container } from "react-bootstrap";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
+export const ProfileView = ({
+  user,
+  token,
+  movies,
+  onUserUpdate,
+  onUserDeregister,
+}) => {
+  const [username, setUsername] = useState(user.username);
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState(user.email);
+  const [birthday, setBirthday] = useState(user.birthday);
+  const navigate = useNavigate();
+
+  // console.log("User's FavoriteMovies:", user.favoriteMovies); // Array of favorite movie IDs
+
+  const favoriteMoviesList =
+    movies && user.favoriteMovies
+      ? movies.filter((m) => user.favoriteMovies.includes(m.id))
+      : [];
+
+  const removeFromFavorites = (movieId) => {
+    axios
+      .delete(
+        `https://flixhive-cf7fbbd939d2.herokuapp.com/users/${user.username}/movies/${movieId}`,
+        {
+          headers: { Authorization: `Bearer ${token}` },
+        }
+      )
+      .then((response) => {
+        console.log("Successfully removed from favorites:", response.data);
+        onUserUpdate(response.data); // Update user data
+      })
+      .catch((error) => {
+        console.log("Error removing movie from favorites", error);
+      });
+  };
+  const handleUpdate = (event) => {
+    event.preventDefault();
+    const updatedUser = { username, password, email, birthday };
+
+    fetch(
```

Comment:
Since you installed `axios` for making API calls, it best practice to be consistent and use it throughout your code.